Assets/MRTK/Core/Utilities/DeviceUtility.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEngine;
+using UnityEngine.XR;
+
+#if UNITY_2019_3_OR_NEWER
+using System.Collections.Generic;
+#endif // UNITY_2019_3_OR_NEWER
+
+namespace Microsoft.MixedReality.Toolkit.Utilities
+{
+    /// <summary>
+    /// Helps with getting data from or about the current HMD or other XR device.
+    /// </summary>
+    public static class DeviceUtility
+    {
+#if UNITY_2019_3_OR_NEWER
+        private static readonly List<XRDisplaySubsystem> XRDisplaySubsystems = new List<XRDisplaySubsystem>();
+#endif // UNITY_2019_3_OR_NEWER
+
+        /// <summary>
+        /// If an HMD is present and running.
+        /// </summary>
+        public static bool IsPresent
+        {
+            get
+            {
+#if UNITY_2019_3_OR_NEWER
+                SubsystemManager.GetInstances(XRDisplaySubsystems);
+                foreach (XRDisplaySubsystem xrDisplaySubsystem in XRDisplaySubsystems)
```

Comment:
Do you know if there's any way of doing this without generating GC each call (i.e. GetInstances I assume has to populate the list with stuff)?

Or is this actually not an issue?

Reply:
Ahh true, after the clear, the subsystems are added to the list. Though, as long as the display subsystems stays the same count, it shouldn't be reallocating, right? `Clear()` without `TrimExcess()` keeps the original capacity I thought, and it looks like the subsystems are [copied in directly](https://github.com/Unity-Technologies/UnityCsReference/blob/61f92bd79ae862c4465d35270f9d1d57befd1761/Modules/Subsystems/SubsystemManager.cs#L35-L44). (Though, if it stays the same count...there's no need to call it multiple times? I'll investigate)

I think that after https://github.com/microsoft/MixedRealityToolkit-Unity/pull/8275 is merged, this can be converted into making a call into that, which should only call `GetInstances` when the current display subsystem stops running (or one never starts up).

Reply:
Reworked this to call into the `XRSubsystemHelpers` class, which handles the lifetime of the cached subsystem better than this code did.